designer/client/src/components/graph/node-modal/fragment-input-definition/item/Item.tsx

```diff
@@ -0,0 +1,84 @@
+import React, { useCallback } from "react";
+import { isEqual } from "lodash";
+import MapKey from "../../editors/map/MapKey";
+import { TypeSelect } from "../TypeSelect";
+import { Validator } from "../../editors/Validators";
+import { Option } from "../FieldsSelect";
+import { FixedValuesPresets, ReturnedType, VariableTypes } from "../../../../../types";
+import SettingsButton from "../buttons/SettingsButton";
+import { FieldsRow } from "../FieldsRow";
+import { Settings } from "../settings/Settings";
+import { useDiffMark } from "../../PathsToMark";
+import { onChangeType, FragmentInputParameter } from "./";
+import { useFieldsContext } from "../NodeRowFields";
+
+interface ItemProps {
+    index: number;
+    item: FragmentInputParameter;
+    validators: Validator[];
+    namespace: string;
+    readOnly?: boolean;
+    showValidation?: boolean;
+    variableTypes: VariableTypes;
+    onChange: (path: string, value: onChangeType) => void;
+    options: Option[];
+    fixedValuesPresets: FixedValuesPresets;
+}
+
+export function Item(props: ItemProps): JSX.Element {
+    const { index, item, validators, namespace, variableTypes, readOnly, showValidation, onChange, options, fixedValuesPresets } = props;
+    const { getIsOpen, toggleIsOpen } = useFieldsContext();
+
+    const isOpen = getIsOpen(item.uuid);
+
+    const path = `${namespace}[${index}]`;
+    const [isMarked] = useDiffMark();
+    const getCurrentOption = useCallback(
+        (typ: ReturnedType | undefined) => {
+            const fallbackValue = { label: typ?.refClazzName, value: typ?.refClazzName };
+            const foundValue = options.find((item) => isEqual(typ?.refClazzName, item.value));
+            return foundValue || fallbackValue;
+        },
+        [options],
+    );
+
+    const openSettingMenu = () => {
+        toggleIsOpen(item.uuid);
+    };
+
+    return (
+        <div>
+            <FieldsRow index={index} uuid={item.uuid}>
```

Comment:
I guess this code is something not related to this change because we just moved it to the separate file and add settings component, but i agree, that this component looks strange here. IMO we shouldn't use `Map` and MapKey components in this context, i will just remove it and paste the content of MapKey directly to the component.

> What is the relation between the parameters in fragmentInputDefinition and Map?

Well, the only relation between these two elements is being the list of the elements and have the name field , but the structure is pretty different. IMO we shouldn't try to make one abstraction for these components, since the case for them is pretty different

Reply:
Changed ✔️